### PR TITLE
Add SelectionResult type to make type alias extraction easier

### LINKED ISSUE #109: Make it easier to extract result types
The inferred types are convenient when the query response is directly used, however it gets a bit complex when I want to extract result type of a query for use elsewhere. 

To exemplify: 

Let's say I have a PostContainer component which fetches list of posts and renders each Post using a PostBody component. Now I need the specific Post type from the query. I have two options: 

1. To use the IPost & IComment types from generated ts file and Pick the fields I need. This obviously works, but is not great because we are now specifying the field selections twice. 
2. Use ResultOf from `@graphql-typed-document-node`: 

```ts
import { query, $ } from "generated/graphql"
import type { ResultOf } from "@graphql-typed-document-node/core";

const fetchPostsQ = query((t) => [
    t.post({ limit: 10, page: $("page"), sort: ['-date_created'] }, (t) => [
        t.title(),
        t.published_at(),
        t.body(),
        t.comments({ limit: 10, filter: { approved_at: { _nnull: true } } }, t => [
                t.user_name(),
                t.content()
        ])
    ]),
]);

type FetchPostsQData = ResultOf<ReturnType<(typeof fetchPostsQ)["toQuery"]>>
type Post = NonNullable<FetchPostsQData["post"]>[number]

```

Option 2 works but is a bit cumbersome. I propose simplifying this to something like: 

```ts
import { query, $, SelectionResult } from "generated/graphql"

const fetchPostsQ = query(...)

type FetchPostsQData = SelectionResult<typeof fetchPostsQ>
type Post = NonNullable<FetchPostsQData["post"]>[number]
```

### PR DESCRIPTION
**Pull request by lorefnon**
Resolves #109

### CODE REVIEW

File: src/Result.ts [hunk 1]
```diff
@@ -81,3 +83,9 @@ type HasInlineFragment<T extends SelectionSet<any> | undefined> = T extends Sele
   : never;
 
 type Merge<M, N> = Omit<M, keyof N> & N;
+
+export type SelectionResult<TSelection> =
```
Comment (timkendall) [inline src/Result.ts:87]: Ah yeah I was planning on actually making this the default for our standard `Result` type. Thoughts on renaming our existing `Result` to `InnerResult` instead (so we can export this higher-level API as `Result`)?

Comment (lorefnon) [reply src/Result.ts:87]: Sure, that sounds better. 

I wanted to avoid a breaking change, but I can update that if you prefer exposing as Result.

Comment (timkendall) [reply src/Result.ts:87]: True. Let's go with this for now.

Comment (timkendall) [reply src/Result.ts:87]: And as always thank you for your contribution!

**Comment by timkendall** [comment]
Published as @timkendall/tql@1.0.0-rc.7, @timkendall/tql@beta, @timkendall/tql-gen@1.0.0-rc.7, @timkendall/tql-gen@latest

**Pull request merged**